### PR TITLE
Skillmap writing guide

### PR DESCRIPTION
**Pull request by ganicke**
Here's a first cut of the a Skillmap writing guide.

### CODE REVIEW

File: docs/writing-docs/skillmaps.md [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+
+```
+# game-maker-guide
+```
+
+Following the heading is a bulleted list of the properties defined for the skillmap.
+
+* **name**: map display name
+* **description**: map description
+* **infoUrl**: ???
```
Comment (ganicke) [inline docs/writing-docs/skillmaps.md:90]: What's this?

File: docs/writing-docs/skillmaps.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+* **backgroundurl**: URL to the background image
+* **bannerurl**: URL to the banner image (displayed in the sidebar when nothing is selected) 
+* **primarycolor**: hex color value for path and locked nodes (slightly translucent)
+* **secondarycolor**: hex color value for unlocked nodes
+* **tertiarycolor**: hex color value for background
+* **highlightcolor**: hex color value for the selected node border and unlocked reward nodes
+* **alternatesources**: ???
```
Comment (ganicke) [inline docs/writing-docs/skillmaps.md:97]: What's this?

File: docs/writing-docs/skillmaps.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+* name: Game Maker Guide
+* description: Level up your game making skills by completing the tutorials in this guide.
+* infoUrl: skillmap/educator-info/int-map-info
+* bannerUrl: /static/skillmap/platformer/activity4.png
+* backgroundurl: /static/skillmap/space/game-maker-guide-background.png
+* primarycolor: #2EA9B0
+* secondarycolor: #F392BD
+* tertiarycolor: #83C252
+* highlightcolor: #FAED28
+* alternatesources: github:https://github.com/microsoft/pxt-skillmap-sample/skillmap.md
```
Comment (ganicke) [inline docs/writing-docs/skillmaps.md:113]: How does this work?

File: docs/writing-docs/skillmaps.md (hunk 1 above)
Comment (shakao) [reply docs/writing-docs/skillmaps.md:99]: ah! this a link to additional educator information (eg a separate markdown file, or another site) about this particular skillmap. we don't have UI exposing it right now, though, so maybe it oculd be left out...

File: docs/writing-docs/skillmaps.md (hunk 2 above)
Comment (shakao) [reply docs/writing-docs/skillmaps.md:106]: this is for migrating progress from one skillmap to another. a user's progress (how much they've unlocked, the projects they've created) is tied to the url of the map, so if you change the url you need to add the old one as an alternate source (comma separated, if there are multiple).

we use it here: https://github.com/microsoft/pxt-arcade/blame/master/docs/skillmap/intermediate-skillmap.md#L11

it's formatted like "github:[path to github map]" or "docs:[path to docs map]"

File: docs/writing-docs/skillmaps.md [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+
+![Skillmap intro graphic](/static/skillmaps/skillmap-intro.jpg)
+
+### ~ hint
+
+#### Sample skillmap
+
+For an example, you can take a look at the [**Skillmap Sample**](https://github.com/microsoft/pxt-skillmap-sample). Use it as a template to start your own!
+
+First though, you should [try out](https://arcade.makecode.com/--skillmap#github:microsoft/pxt-skillmap-sample/skillmap.md) the sample to see how it works.
```
Comment (shakao) [inline docs/writing-docs/skillmaps.md:21]:
```suggestion
First though, you should [try out](https://arcade.makecode.com/--skillmap#github:microsoft/pxt-skillmap-sample/skillmap) the sample to see how it works.
```

(also works with the `.md`, but preferable to specify links without)

File: docs/writing-docs/skillmaps.md [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+
+### ~
+
+## Skillmap paths
+
+A skillmap path has a starting point and a sequence of nodes (activities) to progress through. The activities in nodes along the path are locked until the previous node's an activity is finished.
```
Comment (shakao) [inline docs/writing-docs/skillmaps.md:27]:
```suggestion
A skillmap path has a starting point and a sequence of nodes (activities) to progress through. The activities in nodes along the path are locked until the previous node's activity is finished.
```

File: docs/writing-docs/skillmaps.md [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+
+```
+## space
+```
+
+After the heading is the path property list.
+
+* **name**: path display name
+* **description**: the path description
+* **completionUrl**: the URL path to a completion document (i.e. the "Certificate of Completion")
```
Comment (shakao) [inline docs/writing-docs/skillmaps.md:143]: oh this property should be removed, it's been deprecatd in favor of the completion nodes

File: docs/writing-docs/skillmaps.md [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+
+```
+### space-activity3
+```
+
+The node properties are specified under the `node-id` heading.
+
+* **name**: the display name of the activity node
+* **description**: description of the activity
+* **type**: the type of node, `tutorial` or `certificate`. Currently only the `tutorial` type is used
```
Comment (shakao) [inline docs/writing-docs/skillmaps.md:183]: hmm so basically the setup is: you can specify the `kind` of the node, and each kind  will have "types" (subtypes?) within it. so the default kind of `activity` would theoretically have a set of available types like "tutorial", "game", "design"--but right now we only have "tutorial". and the `completion` kind would have types like "certificate", "assets", "site-theme" (types of prizes), but right now we've only implemented "certificate". 

i think we can leave the `layout` kind out completely for now!

File: docs/writing-docs/skillmaps.md [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+* imageUrl: /static/skillmap/space/enemies.gif
+```
+
+The node `name`, `description`, and `tags` are shown in the skillmap panel.
+
+![skillmap node](/static/skillmaps/skillmap-node.jpg)
+
+### Rewards
+
+The reward node is set automatically at end of the path if the path's `completionUrl` property is set.
```
Comment (shakao) [inline docs/writing-docs/skillmaps.md:212]: we didn't end up doing the automatic inference thing, so content creators have to specify a node with the `completion` kind to render a trophy/reward node

File: docs/writing-docs/skillmaps.md [hunk 9]
```diff
@@ -0,0 +1,222 @@
...
+
+![skillmap reward node](/static/skillmaps/skillmap-reward.jpg)
+
+## Loading a skillmap
+
+Skillmaps are loaded into a MakeCode editor using a URL with the `skillmap` resource type. Using https://arcade.makecode.com/--skillmap will load the default skillmap for MakeCode Arcade. To load skillmaps from somewhere else, you append an external resource address path to this URL. If you want to use a skillmap from the [Skillmap Sample](https://github.com/microsoft/pxt-skillmap-sample) repository in GitHub, then you specify it with a direct path to the skillmap document file.
+
+Connect `https://arcade.makecode.com/--skillmap` with the GitHub resource specifier as a parameter using `#`. So, you have `https://arcade.makecode.com/--skillmap` + `#` + `github:microsoft/pxt-skillmap-sample/skillmap.md`. This forms the full skillmap load URL:
```
Comment (shakao) [inline docs/writing-docs/skillmaps.md:220]:
```suggestion
Connect `https://arcade.makecode.com/--skillmap` with the GitHub resource specifier as a parameter using `#`. So, you have `https://arcade.makecode.com/--skillmap` + `#` + `github:microsoft/pxt-skillmap-sample/skillmap`. This forms the full skillmap load URL:
```

File: docs/writing-docs/skillmaps.md [hunk 10]
```diff
@@ -0,0 +1,222 @@
...
+
+https://arcade.makecode.com/--skillmap#github:microsoft/pxt-skillmap-sample/skillmap.md
```
Comment (shakao) [inline docs/writing-docs/skillmaps.md:222]:
```suggestion
https://arcade.makecode.com/--skillmap#github:microsoft/pxt-skillmap-sample/skillmap
```

**Comment by abchatra** [comment]
@ganicke is this good to merge?

**Comment by ganicke** [comment]
Wow, I think so. All the review feedback was incorporated at the time.

**Pull request merged**